src/client/opamArg.ml

```diff
@@ -894,39 +1214,36 @@ let nonempty_arg_list name doc kind =
   Arg.(non_empty & pos_all kind [] & doc)
 
 (* Common flags *)
-let print_short_flag =
-  mk_flag ["s";"short"]
+let print_short_flag ?(validity=cli_legacy) cli =
```

Comment:
It makes the diff even bigger, but I don't think `validity` should be optional - it runs a risk if `--short` is added elsewhere that ~validity gets forgotten and suddenly a new flag claims to be a 2.0 flag.

Reply:
The alternative for all of these would be for the default to be `OpamCLIVersion.current` - although I still think I'd prefer it to be explicit everywhere, I think we're more likely to notice that an option has _disappeared_ in an older syntax than that it's magically _appeared_!

Reply:
I kept it optional to simplify changes, but yes, it can introduce errors.